src/index/visitor.rs

```diff
@@ -282,29 +283,42 @@ fn visit_data_type(index: &mut Index, type_declatation: &UserTypeDeclaration) {
                     referenced_type: referenced_type.into(),
                 }
             };
-            index.register_type(
-                name.as_ref().unwrap(),
-                type_declatation.initializer.clone(),
-                information,
-            )
+
+            let init = index.add_maybe_constant_expression(type_declatation.initializer.clone());
+            index.register_type(name.as_ref().unwrap(), init, information)
         }
         DataType::ArrayType {
             name,
             referenced_type,
             bounds,
         } => {
-            let dimensions = get_array_dimensions(bounds).unwrap();
+            let dimensions: Result<Vec<Dimension>, CompileError> = bounds
+                .get_as_list()
+                .iter()
+                .map(|it| {
+                    if let AstStatement::RangeStatement { start, end, .. } = it {
+                        Ok(Dimension {
+                            start_offset: index.add_constant_expression(*start.clone()),
```

Comment:
Why can we directly add this as a constant expression but the Strings need special handling?

Reply:
so far - the range statements are not considered at compile time, the expressions passed there, are purely a runtime thing :-/
maybe this will change in the future?